src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -771,9 +771,9 @@ ${if (createTable) {
           ${ qc.generateSQLBody(fakeBody) }"""
         // factor function
         if (func.length == 0) {
-          val funcBody = (fakeBodyAtoms map {x =>
+          val funcBody = (fakeBodyAtoms.zipWithIndex map { case(x, i) =>
           if (ss.variableTableNames contains x.name)
-            s"""${x.name}.R${fakeBody indexOf x}.label"""
+            s"""${if (stmt.head.terms(i).isNegated) "!" else ""}${x.name}.R${fakeBody indexOf x}.label"""
```

Comment:
You could fakeBodyAtoms.zip stmt.head.terms instead of zipWithIndex. Or replace the fakeBody indexOf x with i if that makes sense. Not clear if the order of fakeBodyAtoms has anything to do with stmt.head.terms though. The mobile page is hard to see and they all sound fake.. :/